### DISCUSSION TITLE
GraphRAG Use Cases

### DISCUSSION THREAD

**Discussion by alexchaomander** [microsoft/graphrag#322] [Ideas]
We've seen customers be successful with GraphRAG in areas like news analysis, drug discovery, security, and more.

Share your use cases for GraphRAG here and what features you'd like to see built to best support it!

**Comment by mraguth** [comment]
Happy to explore. Is this package works on structured data like pandas dataframes? And how it differes with other packages like 'llamaindex'? We are heavy users of Llamaindex.

**Comment by kavhad** [comment]
- Audits of various kinds: goverment audits of contractors, producrements. 
- Journalistic research of corruption (e.g. panama papers).

Almost any kind of research could benifit from such systems.

**Comment by silhouettehustler** [comment]
Is it fair to say that any system in need of frequent re-indexing and lots of querying is probably not what GraphRAG was designed for?

**Comment by eddo624** [comment]
I would think it would be cool if you could aggregate all the voting results from the political parties and put them in relation to each other using the GraphRag. That would give more transparency :)

**Comment by ntsarb** [comment]
> We've seen customers be successful with GraphRAG in areas like news analysis, drug discovery, security, and more.
> 
> Share your use cases for GraphRAG here and what features you'd like to see built to best support it!

Support for structured data, e.g. CSV/geoJSON/geoParquet/SQL. I am testing GraphRAG with documents containing references to information found in external CSV files. I could convert these into natural language statements, but this would be very inefficient. 

Think GraphRAG as part of an Agentic AI system wherein, alongside the knowledge graph, there are other forms of structured data, and mechanisms that allow linking/referencing between the data sources.

**Comment by ntsarb** [comment]
> We've seen customers be successful with GraphRAG in areas like news analysis, drug discovery, security, and more.
> 
> Share your use cases for GraphRAG here and what features you'd like to see built to best support it!

A major challenge in building a context-rich knowledgebase is the extraction of context from the source document. We put a title on a document, authors, editors, publisher, publication date, table of contents, and we organise it in chapters, sections, sometimes numbered paragraphs. We use tables, figures, lists, etc. All these represent contextual information that needs to be imported into the knowledge graph.

The DocLing project (also found on GitHub) focuses on extracting text and context (via metadata) from documents of various file formats (e.g. PDF, DOC, PPT, etc). It can use ML and multi-modal LLMs to extract this and produce a structured JSON file as well as image files (for pictures, diagrams, etc).

However, "transferring" this metadata to GraphRAG's knowledge-graph is challenging at the moment. This is because the graph extraction process relies on plaintext (I recently discovered the metadata in JSON/CSV and I am looking into these). 

Hence, I think it may be useful to consider how context-rich documents could be ingested into GraphRAG's knowledge graph more efficiently.

Maybe, by supporting the DocLingDocument format (and corresponding JSON) as input?

**Comment by ntsarb** [comment]
> We've seen customers be successful with GraphRAG in areas like news analysis, drug discovery, security, and more.
> 
> Share your use cases for GraphRAG here and what features you'd like to see built to best support it!

I've been using GraphRAG from the CLI and I wish I could access more detailed information and stats on the LLM's performance. For example, what proportion of LLM responses are thrown away because of errors (e.g. non compliancce with the JSON format)? What proportion of the requests timeout? etc

**Comment by gona-sreelatha** [comment]
We’re using GraphRAG for legacy codebase exploration — specifically to extract requirements, map dependencies, and assist in migrating legacy systems to modern architectures. It’s been a great fit for understanding large and complex software ecosystems.

**Comment by kinthaiofficial** [comment]
GraphRAG's graph-grounded retrieval is particularly powerful for use cases where entity relationships matter more than raw text similarity — traditional vector search finds "nearby text," GraphRAG finds "connected entities."

A few use cases where we've seen it work especially well in agent contexts:

**Organizational knowledge graphs** — when an agent needs to answer "who knows about X, and what did they work on with Y?" traditional RAG fails because the connection isn't in any single document. GraphRAG traverses the relationship.

**Code dependency analysis** — "what would break if I change function X?" requires understanding the call graph, not just finding files that mention X. GraphRAG over a codebase gives agents structural awareness that flat retrieval can't.

**Regulatory/compliance research** — finding which regulations apply to a specific scenario requires traversing the regulatory reference graph (regulation A cites regulation B which applies to entity class C). Community summaries help agents get the right scope before deep retrieval.

**Multi-agent knowledge sharing** — when multiple agents have processed different documents, GraphRAG's entity-centric model means agents can query "what does the system know about entity E?" rather than "what documents mention entity E?" This is closer to how shared knowledge should work in a multi-agent network.

The economic consideration for agent use cases: full GraphRAG (community reports + local/global search) is expensive per query. For agent systems on a budget, local search with selective global search for "what else connects here?" queries is a better cost profile.

We've been integrating knowledge graph retrieval into KinthAI's agent memory architecture: https://blog.kinthai.ai/why-character-ai-forgets-you-persistent-memory-architecture

What's driving your GraphRAG evaluation — enterprise search, research assistance, or code intelligence?

**Comment by raa82** [comment]
One use case I've been working on: regulated industry compliance, specifically energy sector. The challenge there isn't retrieval quality — it's provenance. A compliance officer needs to know not just what the system says but which sources agreed, which conflicted, and what the confidence level actually is. GraphRAG does well on relationship extraction but doesn't track inter-source contradictions. I built a small layer on top for that — happy to share if anyone's working on similar problems: https://github.com/raa82/tekmerdb
